source/concurrent/Promise.ooc

```diff
@@ -36,6 +36,18 @@ _Synchronizer: abstract class {
 			result := this wait(TimeSpan maximumValue)
 		result
 	}
+	wait: static func ~many (items: VectorList<This>) -> Bool {
+		result := true
+		for (i in 0 .. items count)
+			result = result && items[i] wait()
+		result
+	}
+	wait: static func ~timedMany (items: VectorList<This>, time: TimeSpan) -> Bool {
```

Comment:
Oh there was a `TimeSpan maximumValue`. I was going to suggest we implement i. Very good.